codegen/templates/chart/operator-deployment.yamltmpl

```diff
@@ -1,60 +1,94 @@
+{{- $[[ $operatorVar ]]ExtraPodAnnotations := $[[ $operatorVar ]].extraPodAnnotations }}
+{{- $[[ $operatorVar ]]ExtraDeploymentLabels := $[[ $operatorVar ]].extraDeploymentLabels }}
+{{- $[[ $operatorVar ]]ExtraDeploymentAnnotations := $[[ $operatorVar ]].extraDeploymentAnnotations }}
+
   [[- $workloadKind := "Deployment" ]]
   [[- if $operator.Deployment.UseDaemonSet ]]
   [[- $workloadKind = "DaemonSet" ]]
   [[- end -]]
-[[- $operatorVar := (lower_camel $operator.Name) -]]
 
 # [[ $workloadKind ]] manifest for [[ $operator.Name ]]
-{{- $[[ $operatorVar ]] := $.Values.[[ $operatorVar ]]}}
-{{- $[[ $operatorVar ]]Image := $[[ $operatorVar ]].image }}
-{{- $[[ $operatorVar ]]ExtraPodLabels := $[[ $operatorVar ]].extraPodLabels -}}
-{{- $[[ $operatorVar ]]ExtraPodAnnotations := $[[ $operatorVar ]].extraPodAnnotations -}}
-{{- $[[ $operatorVar ]]ExtraDeploymentLabels := $[[ $operatorVar ]].extraDeploymentLabels -}}
-{{- $[[ $operatorVar ]]ExtraDeploymentAnnotations := $[[ $operatorVar ]].extraDeploymentAnnotations -}}
 
 apiVersion: apps/v1
 kind: [[ $workloadKind ]]
 metadata:
   labels:
     app: [[ $operator.Name ]]
+    [[- range $key, $value := $customDeploymentLabels ]]
+    [[ $key ]]: [[ $value ]]
+    [[- end ]]
     {{- range $key, $value := $[[ $operatorVar ]]ExtraDeploymentLabels }}
     {{ $key }}: {{ $value }}
     {{- end }}
-{{- if $[[ $operatorVar ]]ExtraDeploymentAnnotations }}
   annotations:
+    app.kubernetes.io/name: [[ $operator.Name ]]
+    [[- range $key, $value := $customDeploymentAnnotations ]]
+    [[ $key ]]: [[ $value ]]
+    [[- end ]]
     {{- range $key, $value := $[[ $operatorVar ]]ExtraDeploymentAnnotations }}
     {{ $key }}: {{ $value }}
     {{- end }}
-{{- end }}
   name: [[ $operator.Name ]]
   namespace: {{ .Release.Namespace }}
 spec:
   selector:
     matchLabels:
       app: [[ $operator.Name ]]
+      [[- range $key, $value := $customPodLabels ]]
+      [[ $key ]]: [[ $value ]]
+      [[- end ]]
+      {{- range $key, $value := $[[ $operatorVar ]]ExtraPodLabels }}
+      {{ $key }}: {{ $value }}
+      {{- end }}
   template:
     metadata:
       labels:
         app: [[ $operator.Name ]]
+        [[- range $key, $value := $customPodLabels ]]
+        [[ $key ]]: [[ $value ]]
+        [[- end ]]
         {{- range $key, $value := $[[ $operatorVar ]]ExtraPodLabels }}
         {{ $key }}: {{ $value }}
         {{- end }}
       annotations:
+        app.kubernetes.io/name: [[ $operator.Name ]]
+        [[- range $key, $value := $customPodAnnotations ]]
+        [[ $key ]]: [[ $value ]]
+        [[- end ]]
         {{- range $key, $value := $[[ $operatorVar ]]ExtraPodAnnotations }}
         {{ $key }}: {{ $value }}
         {{- end }}
         prometheus.io/path: /metrics
         prometheus.io/port: "9091"
         prometheus.io/scrape: "true"
+        [[- range $key, $value := $customPodAnnotations ]]
+        [[ $key ]]: [[ $value ]]
+        [[- end ]]
```

Comment:
remove this part / it's repeated from above

Reply:
thanks!